dev_tools/auto_merge.py

```diff
@@ -464,57 +458,38 @@ def delete_comment(repo: GithubRepository, comment_id: int) -> None:
                 response.status_code, response.content))
 
 
-def attempt_update_branch_button(pr: PullRequestDetails
-                                 ) -> Union[bool, CannotAutomergeError]:
-    session_cookie = os.getenv(UPDATE_BRANCH_COOKIE_ENV_VARIABLE)
-    if session_cookie is None:
-        return attempt_sync_with_master(pr)
-
-    # Get the pull request page.
-    pr_url = 'https://github.com/{}/{}/pull/{}'.format(
-        pr.repo.organization,
-        pr.repo.name,
-        pr.pull_id)
-    cookies = {'user_session': session_cookie}
-    response = requests.get(pr_url, cookies=cookies)
-    if response.status_code != 200:
-        raise RuntimeError(
-            'Failed to read PR page. Code: {}. Content: {}.'.format(
-                response.status_code, response.content))
+def update_branch(pr: PullRequestDetails) -> Union[bool, CannotAutomergeError]:
+    """Equivalent to hitting the 'update branch' button on a PR.
 
-    # Find the update branch button and relevant tokens.
-    html = response.content.decode()
-    form_guts = re.match(
-        '.*<form class="branch-action-btn'
-        '.*action=".+/pull/.+/update_branch"'
-        '.*<input name="utf8" type="hidden" value="([^"]+)"'
-        '.*<input type="hidden" name="authenticity_token" value="([^"]+)"'
-        '.*<input type="hidden" name="expected_head_oid" value="([^"]+)"'
-        '.*</form>.*', html, re.DOTALL)
-    if form_guts is None:
-        if '(Logged out)' in html:
-            return CannotAutomergeError('Need a fresh :cookie:.')
-        raise RuntimeError(
-            'Failed to find update branch button. Html: {}.'.format(
-                html))
+    As of Feb 2020 this API feature is still in beta. Note that currently, if
+    you attempt to update branch when already synced to master, a vacuous merge
+    commit will be created.
```

Comment:
but this will be squashed out when the PR is merged, yes?

Reply:
Right. Also the method is not called if it isn't needed. This is just a warning for someone looking at the method on its own.